DigitalLearningSolutions.Data.Tests/Services/SectionContentServiceTests.cs

```diff
@@ -993,17 +1058,352 @@ public void Get_section_content_diagnostic_status_is_true_if_one_tutorial_has_di
+                result!.OtherSectionsExist.Should().BeTrue();
+            }
+        }
+
+        [Test]
+        public void Get_section_content_should_not_have_otherSectionsExist_when_only_section_in_application()
+        {
+            // Given
+            const int customisationId = 7967;
+            const int candidateId = 11;
+            const int sectionId = 210;
+
+            // When
+            var result = sectionContentService.GetSectionContent(customisationId, candidateId, sectionId);
+
+            // Then
+            result.Should().NotBeNull();
+            result!.OtherSectionsExist.Should().BeFalse();
+        }
+
+        [Test]
+        public void Get_section_content_should_not_have_otherSectionsExist_when_only_section_not_archived_in_course()
+        {
+            // Given
+            const int customisationId = 21727;
+            const int candidateId = 21727;
+            const int sectionId = 1806;
+
+            // When
+            var result = sectionContentService.GetSectionContent(customisationId, candidateId, sectionId);
+
+            // Then
+            result.Should().NotBeNull();
+            result!.OtherSectionsExist.Should().BeFalse();
+        }
+
+        [Test]
+        public void Get_section_content_should_not_have_otherSectionsExist_when_other_section_is_full_of_archived_tutorials()
+        {
+            using (new TransactionScope())
+            {
+                // Given
+                const int customisationId = 17456;
+                const int candidateId = 210934;
+                const int sectionId = 664;
+
+                // The tutorials of what would be the next section, 668
+                // This is the only other section on this course
+                tutorialContentTestHelper.ArchiveTutorial(2713);
+                tutorialContentTestHelper.ArchiveTutorial(2714);
+                tutorialContentTestHelper.ArchiveTutorial(2715);
+                tutorialContentTestHelper.ArchiveTutorial(2716);
+
+                // When
+                var result = sectionContentService.GetSectionContent(customisationId, candidateId, sectionId);
+
+                // Then
+                result.Should().NotBeNull();
+                result!.OtherSectionsExist.Should().BeFalse();
+            }
+        }
+
+        [Test]
+        public void Get_section_content_should_have_otherSectionsExist_when_other_section_only_has_diagnostic_assessment()
+        {
+            using (new TransactionScope())
+            {
+                // Given
+                const int customisationId = 17456;
+                const int candidateId = 210934;
+                const int sectionId = 664;
+
+                const int hiddenTutorialStatus = 0;
+                const int activeDiagnosticTutorialStatus = 1;
+
+                // The hide but keep the diagnostic assessment of the tutorials in the next section, 668
+                // This is the only other section on this course
```

Comment:
Might be easier to use a course which actually has a section with just a diagnostic assessment, such as 5694? 🙂

Reply:
Yeah I agree! Done for this and the other really long tests in 2137e08 👍